spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/mapper/DefaultColumnMapper.scala

```diff
@@ -102,11 +102,16 @@ class DefaultColumnMapper[T : TypeTag](columnNameOverride: Map[String, String] =
       } yield (getterName, columnRef)
     }.toMap
 
-    // Check if we have all the required columns:
+    /** if this is a column mapper for the top level row, then handle gracefully a set of unmapped columns at the end of
+      * the table columns.  Otherwise, if this is a nested type, then require that there are be no unmapped columns.
+      */
     val mappedColumns = getterMap.values.toSet
     val unmappedColumns = selectedColumns.filterNot(mappedColumns)
-    require(unmappedColumns.isEmpty, s"Columns not found in $tpe: [${unmappedColumns.mkString(", ")}]")
 
+    if (!isTopLevel)
+      require(unmappedColumns.isEmpty, s"Columns not found in nested $tpe: [${unmappedColumns.mkString(", ")}]")
+    else
+      require( selectedColumns.endsWith(unmappedColumns), s"Unmapped columns nust be at end of table definition: [${unmappedColumns.mkString(", ")}]")
```

Comment:
@RussellSpitzer the comment you had here was removed after my most recent push, but I still wanted to address it.  Based on the way that `unmappedColumns` is created (as a filter on `selectedColumns`on line 109), the two sequences should be in the same order.  So, this requirement is essentially asserting that we should map the first _n_ columns from the table with the case class, but subsequent columns can go unmapped.